File: backend/demo/src/app/domain/models/item.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class Item(BaseModel):
    id: UUID
    name: str = Field(..., min_length=1, max_length=255)
    description: str | None = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/demo/src/types/item.ts
export interface Item {
  id: string
  name: string
  description: string | null
  created_at: string
  updated_at: string
}

export interface ItemCreate {
  name: string
  description?: string
}

export interface ItemUpdate {
  name?: string
  description?: string
}

export interface ItemListResponse {
  items: Item[]
  total: number
}
